### PR TITLE
Some new stuff for makefile

### PR DESCRIPTION
**Pull request by LK4D4**
I tried to make it as easy as possible. Without any magic. So at least you need to be aware about go programs layout($GOPATH).
I want to know your thoughts before work further on validate etc.
@crosbymichael @mrunalp

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func main() {
 
 	//allow for relative path for the runC binary
 	if absPath, err := filepath.Abs(os.Args[0]); err != nil {
-		logrus.Fatal("Cannot convert runc path to absolute: %v", err)
+		logrus.Fatalf("Cannot convert runc path to absolute: %v", err)
```
Comment (mrunalp) [inline main.go]: We don't need this anymore because of your other PR.

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,30 @@
-all:
-	go get github.com/tools/godep
+CWD:=$(shell pwd)
+RUNC_TEST_IMAGE=runc_test
+PROJECT=github.com/opencontainers/runc
+TEST_DOCKERFILE=test_Dockerfile
+
+all: godep
 	godep go build -o runc .
```
Comment (mrunalp) [inline Makefile]: Should we document to but the godep binary directory in path or copy it to somewhere in the PATH.

Comment (LK4D4) [inline Makefile]: I dunno, godep target should do it for you.

**Comment by LK4D4** [comment]
@mrunalp I rebased on master.

**Comment by mrunalp** [comment]
@LK4D4 cool, testing it out.

**Comment by mrunalp** [comment]
LGTM

**Comment by mrunalp** [comment]
Needs rebase.

**Comment by LK4D4** [comment]
rebased

**Comment by mrunalp** [comment]
cool, LGTM again ;) I think this is a good place to start.

File: Makefile [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
+runctestimage:
+	docker build -t $(RUNC_TEST_IMAGE) -f $(TEST_DOCKERFILE) .
+
+test: runctestimage
+	docker run --privileged --rm -v $(CWD):/go/src/$(PROJECT) $(RUNC_TEST_IMAGE) make localtest
+
+localtest: godep
+	godep go test -v ./...
+
 install: all
```
Comment (mrunalp) [inline Makefile]: I think we should take out this dependency on all per @crosbymichael comment in the other PR as install requires sudo and could mess things up.

File: Makefile (hunk 2 above)
Comment (tianon) [inline Makefile]: Isn't this `godep` invocation here literally _just_ to set `GOPATH` to include `_workspace`?  Couldn't we just change this `all` target to be simply the following and only download/install/use `godep` for people who actually need to manage the directory contents?
```Makefile
all:
    GOPATH="$$GOPATH:$$PWD/Godeps/_workspace" go build -o runc .
```

Or even more:
```Makefile
export GOPATH=$(GOPATH):$(CURDIR)/Godeps/_workspace

all:
    go build -o runc .

# .....
```

**Comment by LK4D4** [comment]
@tianon changed with gopath

**Comment by tianon** [comment]
:smile: LGTM :+1:

**Comment by mrunalp** [comment]
Could we clean the commit history or squash? We are adding and deleting in the same PR :)

**Comment by LK4D4** [comment]
@mrunalp Yeah, of course, I'll do it later.

**Comment by LK4D4** [comment]
@mrunalp I squashed all commits.

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
Nice! Now we have some way to stand up a CI :)

**Pull request merged**